library/src/hcc_detail/hipsolver.cpp

```diff
@@ -317,4 +319,144 @@ hipsolverStatus_t hipsolverZgetrf(hipsolverHandle_t       handle,
         return rocblas2hip_status(rocsolver_zgetrf_npvt(
             (rocblas_handle)handle, m, n, (rocblas_double_complex*)A, lda, devInfo));
 }
+
+/******************** POTRF ********************/
+hipsolverStatus_t hipsolverSpotrf_bufferSize(
+    hipsolverHandle_t handle, hipsolverFillMode_t uplo, int n, float* A, int lda, int* lwork)
+{
+    size_t sz;
+
+    rocblas_start_device_memory_size_query((rocblas_handle)handle);
+    rocblas_status status = rocsolver_spotrf(
+        (rocblas_handle)handle, hip2rocblas_fill(uplo), n, nullptr, lda, nullptr);
+    rocblas_stop_device_memory_size_query((rocblas_handle)handle, &sz);
+
+    *lwork = (int)sz;
```

Comment:
This function is intended to match the cuSOLVER API (see https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverDN-lt-t-gt-potrf), so changing the type for `lwork` is not an option. I'm not sure what cuSOLVER does if `lwork` exceeds `INT_MAX`, but I'll try and find out so we can adopt the same behaviour. It may be worth noting that this isn't for a batched function, so the 2 GiB is used for a single matrix only.

Reply:
Unfortunately, I haven't been able to find out what cuSOLVER does when `lwork` exceeds `MAX_INT`. I think it should be safe to leave as-is for now, but we should probably investigate further.

Reply:
Could we just return `HIPSOLVER_STATUS_INTERNAL_ERROR` for now? An overflow of a signed integer is undefined behaviour, so leaving it as-is could do anything in the case that lwork exceeds `INT_MAX`. At least if we check for that case, we can define what happens.

Reply:
Sure, I'll add it in.